railties/test/application/test_runner_test.rb

```diff
@@ -249,6 +252,73 @@ def test_generated_scaffold_works_with_rails_test
       assert_match "0 failures, 0 errors, 0 skips", run_test_command('')
     end
 
+    def test_run_multiple_folders
+      create_test_file :models, 'account'
+      create_test_file :controllers, 'accounts_controller'
+
+      run_test_command('test/models test/controllers').tap do |output|
+        assert_match 'AccountTest', output
+        assert_match 'AccountsControllerTest', output
+        assert_match '2 runs, 2 assertions, 0 failures, 0 errors, 0 skips', output
+      end
+    end
+
+    def test_run_multiple_files_and_run_one_file_by_line_only_runs_filter
+      create_test_file :models, 'account'
+      create_test_file :models, 'post'
+
+      run_test_command('test/models/account_test.rb test/models/post_test.rb:5').tap do |output|
+        assert_match '1 runs, 1 assertions', output
+      end
+    end
+
+    def test_multiple_line_filters_only_runs_the_first
+      create_test_file :models, 'account'
+      create_test_file :models, 'post'
+
+      run_test_command('test/models/account_test.rb:4 test/models/post_test.rb:4').tap do |output|
+        assert_match 'AccountTest', output
+      end
+    end
+
+    def test_line_filter_without_line_runs_all_tests
+      create_test_file :models, 'account'
+
+      run_test_command('test/models/account_test.rb:').tap do |output|
+        assert_match 'AccountTest', output
+      end
+    end
+
+    def test_shows_filtered_backtrace_by_default
+      create_backtrace_test
+
+      run_test_command('test/unit/backtrace_test.rb').tap do |output|
+        assert_match '1 runs, 1 assertions, 0 failures', output
+      end
+    end
+
+    def test_backtrace_option
+      create_backtrace_test
+
+      run_test_command('test/unit/backtrace_test.rb -b').tap do |output|
+        assert_match '1 runs, 1 assertions, 0 failures', output
+      end
+
+      run_test_command('test/unit/backtrace_test.rb --backtrace').tap do |output|
+        assert_match '1 runs, 1 assertions, 0 failures', output
+      end
+    end
+
+    def test_show_full_backtrace_using_backtrace_environment_variable
+      create_backtrace_test
+
+      switch_env 'BACKTRACE', 'true' do
+        run_test_command('test/unit/backtrace_test.rb').tap do |output|
+          assert_match '1 runs, 1 assertions, 0 failures', output
+        end
+      end
+    end
+
     private
```

Comment:
The way I see it, the happy path testing is to make sure we're correctly funneling to Minitest. Thus adding tests for more than our extensions would be testing Minitest.